### PR TITLE
Bug in DrILL export

### LINKED ISSUE #30813: Export fails on drill
Exporting in drill fails under certain circumstances.
This could be related to the fact, that the ws is a group, so you can't get its history directly.
Also there must be a check if history exists, and if it is not empty.
In any case failure of exports shouldn't result into crying mantid.

### Steps to reproduce the behavior
Start drill. Set datasearch directory to current cycle internalUse of D11.
Put 000560 as sample, check OutputPanels in global settings.
Click process > 💣

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

During the automatic export in DrILL, criteria are applied on workspaces to verify that a specific export algorithm can be applied on the data. This is using the processing parameters found in the workspace history. This was causing a crash with workspaces with empty history (for example, panel workspaces in D11 data processing).

To resolve this problem, the algorithm history is now in a try-catch block and the criteria are validated once per sample on the output workspace.

**To test:**

Try to reproduce the issue #30813 

Fixes #30813 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by gvardany — CHANGES_REQUESTED** [review]
There is an issue still. Try the following:
Enable all the savers, set output panels and a few wedges.
Put any sample and run. Now it crashes even harder.
We also need a test in this kind of most inclusive scenario.

File: scripts/Interface/ui/drill/model/DrillExportModel.py [hunk 1]
```diff
@@ -178,34 +179,50 @@ def _logSuccessExport(self, wsName):
...
+        if not exportPath:
+            exportPath = os.getcwd()
         workspaceName = sample.getOutputName()
 
+        try:
+            outputWsGroup = mtd[workspaceName]
+            names = outputWsGroup.getNames()
+            outputWs = names[0]
+        except:
+            return
```
Comment (gvardany) [inline scripts/Interface/ui/drill/model/DrillExportModel.py:201]: Can we log a useful message to notice?

Comment (gvardany) [reply scripts/Interface/ui/drill/model/DrillExportModel.py:201]: As discussed offline, logging here is not critical.

**Review by gvardany — DISMISSED** [review]
Looks good, it is better to check the compatibility criteria only once per sample. This revealed another issue in Load/Save NexusProcessed which will be handled separately https://github.com/mantidproject/mantid/issues/30875

File: scripts/Interface/ui/drill/model/DrillExportModel.py [hunk 2]
```diff
@@ -178,34 +179,50 @@ def _logSuccessExport(self, wsName):
...
-            workspaceName (str): name of the workspace
+            sample (DrillSample): sample to be exported
         """
         exportPath = config.getString("defaultsave.directory")
+        if not exportPath:
+            exportPath = os.getcwd()
         workspaceName = sample.getOutputName()
 
+        try:
+            outputWsGroup = mtd[workspaceName]
```
Comment (gvardany) [inline scripts/Interface/ui/drill/model/DrillExportModel.py:197]: I just realized that output workspace doesn't have to be a workspace group. I think we should check that too.

Comment (gui-co) [reply scripts/Interface/ui/drill/model/DrillExportModel.py]: I added a check for that.

**Review by gvardany — APPROVED** [review]
This is good to go now.

**Pull request merged**